ui/perfherder/graphs/GraphsViewControls.jsx

```diff
@@ -39,9 +46,16 @@ export default class GraphsViewControls extends React.Component {
       testData,
     } = this.props;
 
+    const measurementUnits = this.extractMeasurementUnitsSet(testData);
+
     return (
       <Container fluid className="justify-content-start">
-        <TestDataModal showModal={showModal} toggle={toggle} {...this.props} />
+        <TestDataModal
+          showModal={showModal}
+          toggle={toggle}
+          plottedUnits={measurementUnits}
```

Comment:
The `TestDataModal` already has access to the `testData` prop, so you could move this value into the modal rather than passing it in.

Reply:
Initially I wanted to do this, but noticed the `GraphsContainer` sibling required the same `measurementUnits` value. I opted to pre compute it once in the `GraphViewControls` parent & provide it to both `TestDataModal` & `GraphsContainer`, so they don't recalculate the same thing twice.